File: output/ChatGPT/codenet/C++/Python/s525470559.py
task = {
    "name": "",
    "time": 0
}

n, q = map(int, input().split())

Q = []

for i in range(n):
    tmp = task.copy()
    tmp["name"], tmp["time"] = input().split()
    tmp["time"] = int(tmp["time"])
    Q.append(tmp)

elaps = 0

while len(Q) > 0:
    p = Q.pop(0)
    if p["time"] <= q:
        elaps += p["time"]
        print(p["name"], elaps)
    else:
        p["time"] -= q
        elaps += q
        Q.append(p)

File: output/ChatGPT/codenet/C++/Java/s525470559.java
import java.util.LinkedList;
import java.util.Queue;
import java.util.Scanner;

class Task {
    String name;
    int time;
}

public class s525470559 {
    public static void main(String[] args) {
        int n, q;
        int elaps = 0;
        Queue<Task> Q = new LinkedList<>();

        Scanner scanner = new Scanner(System.in);
        n = scanner.nextInt();
        q = scanner.nextInt();

        for (int i = 0; i < n; i++) {
            Task tmp = new Task();
            tmp.name = scanner.next();
            tmp.time = scanner.nextInt();
            Q.add(tmp);
        }

        while (!Q.isEmpty()) {
            Task p = Q.poll();
            if (p.time <= q) {
                elaps += p.time;
                System.out.println(p.name + " " + elaps);
            } else {
                p.time -= q;
                elaps += q;
                Q.add(p);
            }
        }
    }
}